packages/default/scss/switch/_layout.scss

```diff
@@ -1,99 +1,131 @@
-$switch-size: 6em !default;
-$switch-handle-size: (.4 * $switch-size) !default;
-
 @include exports('switch/layout') {
-    $switch-border-width: 1px;
-    $switch-padding: 2 * $switch-border-width;
 
     .k-switch {
+        @include border-radius( $switch-border-radius );
         cursor: pointer;
+        width: $switch-size;
+        background: none;
         border: 0;
-        display: inline-block;
-        font-size: $font-size-xs;
+        outline: 0;
+        box-shadow: none;
+        display: inline-flex;
         overflow: hidden;
-        position: relative;
-        text-align: left;
-        user-select: none;
         vertical-align: middle;
-        width: $switch-size;
-        outline: 0;
+        font-size: $font-size-xs;
+        user-select: none;
+        text-align: left;
 
         [type='checkbox'] {
             display: none;
         }
     }
 
     .k-switch,
-    .k-switch-wrapper,
     .k-switch-container,
     .k-switch-handle {
         box-sizing: border-box;
     }
 
-    .k-switch-wrapper {
-        display: none; // TODO: do not render
-    }
-
-    .k-switch .k-switch-background {
-        border-top-right-radius: 0;
-        border-bottom-right-radius: 0;
-    }
-
     .k-switch-container {
-        padding: $switch-padding 0;
-        display: block;
-        width: 100%;
-        background: transparent;
-        transform: translateZ(0);
+        @include border-radius( $switch-border-radius );
+        padding: $switch-container-padding-y $switch-container-padding-x;
+        border-width: $switch-container-border-width;
+        border-style: solid;
+        flex: 1 0 auto;
+        position: relative;
         outline: 0;
+        transition: background-color 200ms ease-out 0s;
     }
 
     .k-switch-handle {
-        position: relative;
+        @include border-radius( $switch-handle-border-radius );
         width: $switch-handle-size;
         height: $switch-handle-size;
-        display: inline-block;
-        margin: 0 ($switch-padding * 3) 0 $switch-padding;
-        border-width: $switch-border-width;
+        border-width: $switch-handle-border-width;
         border-style: solid;
+        display: inline-block;
         vertical-align: middle;
```

Comment:
not needed

Reply:
When removed, the handle is no longer aligned correctly.